### PR TITLE
feat: introduce large cache for swr

### PR DESCRIPTION
**Pull request by sjaanus**
Previously, clearing the SWR cache cleared all entries. Now you can configure the cache size.

1. This makes the search more fluid. Previously, if you went back and forth on pages, you were always sent to the loading state. 
2. This also solves the issue where the command bar search cleared the cache for all other searches. 
3. Additionally, it addresses the problem where the global search cleared the cache for project search.

### CODE REVIEW

File: frontend/src/hooks/api/getters/useFeatureSearch/useFeatureSearch.test.tsx [hunk 1]
```diff
@@ -42,26 +42,4 @@ describe('useFeatureSearch', () => {
         await screen.findByText(/Features:/);
         await screen.findByText(/Total:/);
     });
-
-    test('should keep only latest cache entry', async () => {
```
Comment (kwasniew) [inline frontend/src/hooks/api/getters/useFeatureSearch/useFeatureSearch.test.tsx:46]: what is this test (hook inside a component usage) replaced with? I only see low level hook tests but usually the most revealing tests for hooks misbehavior are usage in component tests

File: frontend/src/hooks/useClearSWRCache.ts [hunk 2]
```diff
@@ -1,14 +1,38 @@
...
+
+interface Cache<Data = any> {
+    keys(): IterableIterator<string>;
+    get(key: string): State<Data> | undefined;
+    set(key: string, value: State<Data>): void;
+    delete(key: string): void;
+}
+
+export const manageCacheEntries = (
+    cache: Cache,
```
Comment (kwasniew) [inline frontend/src/hooks/useClearSWRCache.ts:11]: why do we have generic in Cache while we skip it here and default to any?

File: frontend/src/hooks/useClearSWRCache.ts [hunk 3]
```diff
@@ -1,14 +1,38 @@
...
+import { type State, useSWRConfig } from 'swr';
+
+interface Cache<Data = any> {
+    keys(): IterableIterator<string>;
+    get(key: string): State<Data> | undefined;
+    set(key: string, value: State<Data>): void;
+    delete(key: string): void;
+}
+
+export const manageCacheEntries = (
```
Comment (kwasniew) [inline frontend/src/hooks/useClearSWRCache.ts:10]: what are you managing about the cache? manage is a very generic name that is not intention revealing

Comment (kwasniew) [reply frontend/src/hooks/useClearSWRCache.ts:10]: is it clearCacheEntries?

File: frontend/src/hooks/api/getters/useFeatureSearch/useFeatureSearch.test.tsx (hunk 1 above)
Comment (sjaanus) [reply frontend/src/hooks/api/getters/useFeatureSearch/useFeatureSearch.test.tsx:46]: Pulled this test back, but altered a little.

File: frontend/src/hooks/useClearSWRCache.ts (hunk 3 above)
Comment (sjaanus) [reply frontend/src/hooks/useClearSWRCache.ts:10]: Yes

**Review by kwasniew — APPROVED** [review]

**Pull request merged**